app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers.rb

```diff
@@ -3,47 +3,92 @@ class ManageIQ::Providers::IbmCloud::Inventory::Collector::PowerVirtualServers <
   require_nested :NetworkManager
   require_nested :StorageManager
 
-  def connection
-    @power_iaas ||= manager.connect(:service => "PowerIaas")
+  def pvm_instances
+    @pvm_instances ||= pvm_instances_api.pcloud_pvminstances_getall(cloud_instance_id).pvm_instances || []
   end
 
-  def vms
-    connection.get_pvm_instances
+  def pvm_instance(instance_id)
+    pvm_instances_api.pcloud_pvminstances_get(cloud_instance_id, instance_id)
   end
 
   def image(img_id)
-    connection.get_image(img_id)
+    images_api.pcloud_cloudinstances_images_get(cloud_instance_id, img_id)
   end
 
   def images
-    connection.get_images
+    @images ||= images_api.pcloud_cloudinstances_images_getall(cloud_instance_id).images || []
   end
 
   def volumes
-    connection.get_volumes
+    @volumes ||= volumes_api.pcloud_cloudinstances_volumes_getall(cloud_instance_id).volumes || []
   end
 
   def volume(volume_id)
-    connection.get_volume(volume_id)
+    volumes_api.pcloud_cloudinstances_volumes_get(cloud_instance_id, volume_id)
   end
 
   def networks
-    connection.get_networks
+    @networks ||= networks_api.pcloud_networks_getall(cloud_instance_id).networks || []
+  end
+
+  def network(network_id)
+    networks_api.pcloud_networks_get(cloud_instance_id, network_id)
   end
 
   def ports(network_id)
-    connection.get_network_ports(network_id)
+    @ports ||= networks_api.pcloud_networks_ports_getall(cloud_instance_id, network_id).ports || []
   end
 
   def sshkeys
-    connection.get_ssh_keys
+    @ssh_keys ||= tenants_api.pcloud_tenants_get(tenant_id).ssh_keys
   end
 
-  def system_pool
-    connection.get_system_pool.values
+  def system_pools
+    @system_pools ||= system_pools_api.pcloud_systempools_get(cloud_instance_id)
   end
 
   def storage_types
-    connection.get_storage_types
+    # TODO
+    # connection.get_storage_types
+    []
+  end
+
+  private
+
+  def connection
+    @connection ||= manager.connect
+  end
+
+  def cloud_instance_id
+    @cloud_instance_id ||= manager.uid_ems
+  end
+
+  def tenant_id
+    # TODO
```

Comment:
Do we need to generate the entire [IBM Cloud Resource Controller API](https://cloud.ibm.com/apidocs/resource-controller/resource-controller) to grab tenant id and region?

Reply:
Already done, I'm using it [here](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/48/files#diff-b636667a0f7f3e051ca4608c0ab24a4bR126) I'm just working on a clean way of returning it from connect.

I had started using the api_client instance as the return from connect on the vpc side because that's all that was needed to pass in to the other apis but I'll need to modify that for the power side